### PR TITLE
Strict mode

### PR DESCRIPTION
**Pull request by ftreguer**
- Enable TS strict mode
- Use ts-sdk 2.0.0-alpha.0 with strict mode and new UNSClient constructor/init

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
 import { handleErrors, handleFetchError } from "./errorHandler";
+import { Token, Transaction } from "./types";
 import * as UTILS from "./utils";
 import { getUrlOrigin } from "./utils";
 
 export class UNSCLIAPI {
     public network: any;
 
-    constructor(networkPreset: any, customNodeUrl?: string) {
+    public setConfig(networkPreset: any, customNodeUrl?: string) {
```
Comment (Nigui) [inline src/api.ts]: it could be consistent with sdk if named `init`

File: src/api.ts [hunk 2]
```diff
@@ -53,7 +54,10 @@ export class UNSCLIAPI {
      * @param transactionId
      * @param msdelay
      */
-    public async getTransaction(transactionId: string, msdelay: number = 0): Promise<any> {
+    public async getTransaction(
+        transactionId: string,
+        msdelay: number = 0,
+    ): Promise<Transaction & { chainmeta: ChainMeta }> {
```
Comment (Nigui) [inline src/api.ts]: You could declare a type `type WithChainMeta<T> = T & { chainmeta: ChainMeta }` then you would have : 
- `Promise<WithChainMeta<Transaction>>`
- `Promise<WithChainMeta<Unik>>`
- `Promise<WithChainMeta<Wallet>>`
- and so on

File: src/api.ts [hunk 3]
```diff
@@ -125,7 +129,9 @@ export class UNSCLIAPI {
      *
      * @param unikid Get UNIK token properties
      */
-    public async getUnikProperties(unikid: string): Promise<any> {
+    public async getUnikProperties(
+        unikid: string,
+    ): Promise<{ data: Array<{ [_: string]: string }>; chainmeta: ChainMeta }> {
```
Comment (Nigui) [inline src/api.ts]: can't you reuse `Response` type from SDK

File: src/api.ts (hunk 2 above)
Comment (Nigui) [reply src/api.ts]: or use `@ts-sdk` `ReponseWithChainMeta`

File: src/baseCommand.ts [hunk 4]
```diff
@@ -185,7 +204,11 @@ export abstract class BaseCommand extends Command {
         );
     }
 
-    protected async withAction<T>(actionDescription: string, callback, ...args): Promise<T> {
+    protected async withAction<T>(
+        actionDescription: string,
+        callback: (...args: any[]) => Promise<any> | any,
```
Comment (Nigui) [inline src/baseCommand.ts]: returns `Promise<any>` here has no effect because of following `any`

File: src/commands/read-wallet.ts [hunk 5]
```diff
@@ -33,9 +35,9 @@ export class ReadWalletCommand extends ReadCommand {
     }
 
     protected async do(flags: Record<string, any>, args?: Record<string, any>): Promise<NestedCommandOutput> {
-        const walletId = args.walletId;
+        const walletId = args?.walletId;
         const wallet: any = await this.api.getWallet(walletId);
-        const tokens: any = await this.api.getWalletTokens(walletId);
+        const tokens: { data: Token[]; chainmeta: ChainMeta } | void = await this.api.getWalletTokens(walletId);
```
Comment (Nigui) [inline src/commands/read-wallet.ts]: how it could returns `void` ?
Hint: I tried locally without `void` and guess what...

**Review by Nigui — APPROVED** [review]

File: src/api.ts (hunk 3 above)
Comment (ftreguer) [reply src/api.ts]: WithChainMeta like upper comment

File: src/baseCommand.ts (hunk 4 above)
Comment (ftreguer) [reply src/baseCommand.ts]: Hehe your're right!

File: src/commands/read-wallet.ts (hunk 5 above)
Comment (ftreguer) [reply src/commands/read-wallet.ts]: old ugly code

**Pull request merged**